### PR TITLE
ReactNative Support for External Storage

### PR DESCRIPTION
**Pull request by bhariharan**
Goes with the corresponding ```shared``` repo commit.

### CODE REVIEW

File: libs/SalesforceHybrid/src/com/salesforce/androidsdk/phonegap/plugin/SmartStorePlugin.java [hunk 1]
```diff
@@ -36,8 +36,8 @@
 import com.salesforce.androidsdk.smartstore.store.QuerySpec.QueryType;
 import com.salesforce.androidsdk.smartstore.store.SmartStore;
 import com.salesforce.androidsdk.smartstore.store.SmartStore.SmartStoreException;
-import com.salesforce.androidsdk.smartstore.store.StoreCursor;
 import com.salesforce.androidsdk.smartstore.store.SoupSpec;
+import com.salesforce.androidsdk.smartstore.store.StoreCursor;
```
Comment (bhariharan) [inline libs/SalesforceHybrid/src/com/salesforce/androidsdk/phonegap/plugin/SmartStorePlugin.java]: Took the opportunity to fix the import ordering.

File: native/NativeSampleApps/SmartSyncExplorer/src/com/salesforce/samples/smartsyncexplorer/ui/MainActivity.java [hunk 2]
```diff
@@ -70,6 +66,10 @@
 import com.salesforce.samples.smartsyncexplorer.loaders.ContactListLoader;
```
Comment (bhariharan) [inline native/NativeSampleApps/SmartSyncExplorer/src/com/salesforce/samples/smartsyncexplorer/ui/MainActivity.java]: Took the opportunity to fix the import ordering.

File: libs/SalesforceReact/src/com/salesforce/androidsdk/reactnative/bridge/SmartStoreReactBridge.java [hunk 3]
```diff
@@ -518,14 +536,39 @@ public void getSoupIndexSpecs(ReadableMap args,
...
 
     /**
+     * Native implementation of getSoupIndexSpecs
+     * @param args
+     * @param successCallback
+     * @param errorCallback
+     * @return
+     */
+    @ReactMethod
+    public void getSoupSpec(ReadableMap args, final Callback successCallback,
```
Comment (bhariharan) [inline libs/SalesforceReact/src/com/salesforce/androidsdk/reactnative/bridge/SmartStoreReactBridge.java]: New method.

File: libs/SalesforceReact/src/com/salesforce/androidsdk/reactnative/bridge/SmartStoreReactBridge.java [hunk 4]
```diff
@@ -285,21 +284,42 @@ public void upsertSoupEntries(ReadableMap args,
 	 * @return
 	 */
 	@ReactMethod
-	public void registerSoup(ReadableMap args,
-                             final Callback successCallback, final Callback errorCallback){
+	public void registerSoup(ReadableMap args, final Callback successCallback,
```
Comment (bhariharan) [inline libs/SalesforceReact/src/com/salesforce/androidsdk/reactnative/bridge/SmartStoreReactBridge.java]: Reworked this method to support the new arguments. Same as the ```Cordova``` plugin implementation.

File: libs/SalesforceReact/src/com/salesforce/androidsdk/reactnative/bridge/SmartStoreReactBridge.java [hunk 5]
```diff
@@ -518,14 +536,39 @@ public void getSoupIndexSpecs(ReadableMap args,
...
-		}
-		catch (JSONException e) {
+		} catch (JSONException e) {
 			Log.e(LOG_TAG, "getSoupIndexSpecs", e);
 			errorCallback.invoke(e.toString());
 		}
 	}
 
     /**
+     * Native implementation of getSoupIndexSpecs
```
Comment (kchitalia) [inline libs/SalesforceReact/src/com/salesforce/androidsdk/reactnative/bridge/SmartStoreReactBridge.java]: This should reflect the correct method name

**Comment by kchitalia** [comment]
Looks good 👍

**Pull request merged**